substrate/frame/broker/src/dispatchable_impls.rs

```diff
@@ -437,4 +438,39 @@ impl<T: Config> Pallet<T> {
 		Self::deposit_event(Event::AllowedRenewalDropped { core, when });
 		Ok(())
 	}
+
+	pub(crate) fn do_swap_leases(id: TaskId, other: TaskId) -> DispatchResult {
+		let mut id_leases_count = 0;
+		let mut other_leases_count = 0;
+		let leases = Leases::<T>::get()
+			.into_inner()
+			.into_iter()
+			.map(|mut lease| {
+				if lease.task == id {
+					lease.task = other;
+					id_leases_count += 1;
+				} else if lease.task == other {
+					lease.task = id;
+					other_leases_count += 1;
+				}
+				lease
+			})
+			.collect::<Vec<_>>();
+
+		// Ensure both leases exist
+		ensure!(id_leases_count > 0 && other_leases_count > 0, Error::<T>::UnknownReservation);
+		// And ensure there are just to leases to be swapped
+		ensure!(id_leases_count == 1 && other_leases_count == 1, Error::<T>::TooManyLeases);
```

Comment:
This is wrong. When one of the leases for example already ended, you should still be able to swap it with a "working one"

Reply:
I can imagine the source lease (id in this case) might not exist and is not correct to enforce its existence because of what you wrote.

But why would one want to swap something with a non-existent lease? The way I understand it we should have:

```
ensure!(other_leases_count > 0, Error::<T>::UnknownXXX);
```

Reply:
Never mind, it's unnecessary complication. Even if someone does it - he/she can revert it back anytime.